### PR TITLE
remove explicit LangVersion from a few projects

### PR DESCRIPTION
**Pull request by kasperk81**

### CODE REVIEW

**Comment by ViktorHofer** [comment]
This one can be removed as well: https://github.com/dotnet/runtime/blob/16cb35fda05165d679603c1c9c7ca5710e9bd3df/src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj#L10.

**Comment by kasperk81** [comment]
i left it because it targets multiple frameworks https://github.com/dotnet/runtime/blob/16cb35fda05165d679603c1c9c7ca5710e9bd3df/src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj#L3

should be ok to remove it now.

**Comment by ViktorHofer** [comment]
As explained [here](https://github.com/dotnet/runtime/pull/75168#discussion_r967084597) it can be removed now.

**Comment by kasperk81** [comment]
all checks passed

File: src/libraries/Common/tests/StaticTestGenerator/StaticTestGenerator.csproj [hunk 1]
```diff
@@ -1,9 +1,8 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
```
Comment (ViktorHofer) [inline src/libraries/Common/tests/StaticTestGenerator/StaticTestGenerator.csproj:4]: Revert. This project doesn't import any of the repo infrastructure.

File: src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
   <PropertyGroup>
       <OutputType>Exe</OutputType>
-      <TargetFramework>net6.0</TargetFramework>
-      <LangVersion>latest</LangVersion>
+      <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
```
Comment (ViktorHofer) [inline src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj:5]: Again, please revert this change. This project doesn't import any of the repo's msbuild infrastructure.

Comment (kasperk81) [reply src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj:5]:
```suggestion
      <TargetFramework>net7.0</TargetFramework>
```

Comment (ViktorHofer) [reply src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj]: That patch should be removed as the library isn't part of the repo's build.

Comment (kasperk81) [reply src/libraries/System.Text.RegularExpressions/tools/GenerateRegexCasingTable.csproj]: it is removed from installer and we wont be needing any patchs with these changes.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Thanks a lot. Great work!

**Comment by kasperk81** [comment]
@ViktorHofer looks good to merge?

**Comment by ViktorHofer** [comment]
Thanks a lot
